src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1449,8 +1447,14 @@ private bool CheckEventValueKind(BoundEventAccess boundEvent, BindValueKind valu
                     }
                 }
 
-                return true;
             }
+
+            if (RequiresVariableReceiver(receiver, eventSymbol) && !CheckIsValidReceiverForVariable(eventSyntax, receiver, valueKind, diagnostics))
```

Comment:
>if (RequiresVariableReceiver(receiver, eventSymbol) && !CheckIsValidReceiverForVariable(eventSyntax, receiver, valueKind, diagnostics)) [](http://example.com/codeflow?start=12&length=135)

It looks like we are duplicating this check for ```valueKind != BindValueKind.CompoundAssignment``` code path. Is there a specific reason why we are not doing this the `if` block above?

Reply:
No, it's fine to have in the `if`. I'll move it back